src/conduits/java/com/enderio/conduits/common/network/ConduitSavedData.java

```diff
@@ -63,24 +67,47 @@ private ConduitSavedData(CompoundTag nbt) {
 
 
                     for (Tag tag2 : graphObjectsTag) {
-                        NodeIdentifier node = new NodeIdentifier(BlockPos.of(((LongTag)tag2).getAsLong()));
+                        BlockPos pos = BlockPos.of(((LongTag)tag2).getAsLong());
+                        NodeIdentifier node = new NodeIdentifier(pos);
                         graphObjects.add(node);
+                        putNodeIdentifier(value, pos, node);
                     }
 
                     for (Tag tag2: graphConnectionsTag) {
                         CompoundTag connectionTag = (CompoundTag) tag2;
                         connections.add(new Pair<>(graphObjects.get(connectionTag.getInt("0")), graphObjects.get(connectionTag.getInt("1"))));
                     }
+
                     NodeIdentifier graphObject = graphObjects.get(0);
                     Graph.integrate(graphObject, List.of());
                     merge(graphObject, connections);
+
                     networks.get(value).add(graphObject.getGraph());
                 }
             }
         }
         EnderIO.LOGGER.info("Conduit network deserialization finished, took {}ms", System.currentTimeMillis() - start);
     }
 
+    // Serialization
+
+    /* NBT layout
+    data
+      ┖ graphs (list)
+          ┖ [index]
+              ┠ graphs (list) // One type of conduits' graphs
+              ┃   ┠ [index]
+              ┃   ┃   ┠ graphConnections (list)
+              ┃   ┃   ┃   ┖ [index]
+              ┃   ┃   ┃       ┠ 0: [first object's index]
+              ┃   ┃   ┃       ┖ 1: [second object's index]
+              ┃   ┃   ┖ graphObjects (list)
+              ┃   ┃       ┖ [index]: long (representing BlockPos.asLong())
+              ┃   ┖ [next index]
+              ┃       ┖ ...
+              ┖ type: [conduit type] (ex. "enderio:power3")
```

Comment:
instead of index write element and move type above graphs as that is a smaller element and might be overlocked otherwise